Alex k ino/triage list client list

AlexK-ino opened the pull request:
## Description of the change
Added:

- customized progress bar
- tooltip box
- customized checkbox
- button "Snooze"
- mock data with American format date

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #7 

## Checklists

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

Wmaileq commented on .eslintignore:
```diff
@@ -1,2 +1,3 @@
-react-app-env.d.ts
\ No newline at end of file
+react-app-env.d.ts
+reportWebVitals.ts
```
Add the new line

Wmaileq commented on public/index.html:
```diff
@@ -17,7 +17,7 @@
     <title>Case Triage Tool</title>
   </head>
-  <body>
+  <body style="margin: 0;">
```
Use global styles in favor of inline styles

Wmaileq commented on src/components/Checkbox.tsx:
```diff
@@ -0,0 +1,82 @@
+  width: 16px;
+  height: 16px;
+  border-radius: 3px;
+  display: ${(props: Props) => (props.checked ? "flex" : "none")};
```
Use `props destructuring`.

Wmaileq commented on src/components/ClientCard.tsx:
```diff
@@ -0,0 +1,303 @@
+    });
+  };
+
+  const setProgressColor = () => {
```
`getProgressColor`
Also, as an alternative, you could do it as `IIFE` or `useMemo` and name it just `progressColor`.

Wmaileq commented on src/components/IndividualCards.tsx:
```diff
@@ -14,28 +14,43 @@
-  </Container>
+export const IndividualCards = () => {
+  const [users, setUsers] = useState(mockData);
+  const handleChangeUser = (val: User) => {
```
Maybe just `changeUser` or `setUser`, or `setUserData`?

Wmaileq commented on src/components/ClientCard.tsx:
```diff
@@ -0,0 +1,303 @@
+
+const Box = styled.div`
+  display: block;
+`;
```
Please, add new lines between function declarations and before return operator. It looks too dense now.

Also, maybe separate styles into a different file and name it `*.styles.tsx`, thoughts?

Wmaileq commented on src/App.test.tsx:
```diff
@@ -16,7 +16,7 @@
 import React from "react";
 import { render, screen } from "@testing-library/react";
-import App from "./App";
+import { App } from "./App";
```
Prefer `default export`

Wmaileq commented on src/components/ClientCard.tsx:
```diff
@@ -0,0 +1,303 @@
+  progress: number;
+  snoozed: boolean;
+  checked: boolean;
+  expanded: boolean;
```
Boolean constants should start with `is`, `isExanded`, `isChecked`

Wmaileq commented on src/components/ClientCard.tsx:
```diff
@@ -0,0 +1,303 @@
+  outline: 0;
+`;
+export const ClientCard = ({ user, changeUser }: Props) => {
+  const handleChangeChecked = (
```
What is the mission of `handle` keyword here? Maybe name these functions `toggleIsChecked`, `toggleIsExpanded`, ...?

Wmaileq reviewed (changes requested):
Initial review

jessex reviewed (changes requested):
I tried to start the app locally via `yarn start` to test this and it failed to compile (see screenshot below). Is that happening for you, @AlexK-ino? I know we had a similar issue on an earlier PR where @ohaibbq and I could not start it locally, so if you're not trying to run this locally via `yarn start`, you definitely should do so before issuing PRs.

ohaibbq commented on .github/pull_request_template.md:
```diff
@@ -17,13 +17,13 @@
 ### Code review
 
-These boxes should be checked by reviewers prior to merging:
+These boxes should be isChecked by reviewers prior to merging:
```
I don't think you meant to find/replace this.

ohaibbq commented:
Thanks, Alex! Its looking good. Some styling issues to fix up before I review the code.

Please reference the checklist with my attached annotated image.

- [ ] 1. The font color and the circle fill color is based on what % full the circle is. Please see the mocks for the appropriate colors for the various circle states. (0,1,2,3,4,5,6,7). The circle height should be as tall as the 2 lines of text to the right
- [ ] 2. There is too much whitespace between the circle and the first column.
- [ ] 3. There is too much whitespace between the two lines of text
- [ ] 4. The "tooltip" text does vertical alignment does not match the alignment specified in the mocks
- [ ] 5. The left side of the exclamation icon should align with the left side of the circle 
- [ ] 6. On smaller screen sizes, an odd left border appears on the list item
- [ ] 7. When hovering an ID link, the text falls out of left alignment with the line above
- [ ] 8. The tooltip is not activated via click. It should be shown by default and is persistent
- [ ] 9. The tooltip is missing the arrow icon on the right side
- [ ] 10. The bottom border list items needs to be consistent. There is currently an inconsistency based on whether or not item has a tooltip
- [ ] 11. Please allow for some dates to have an red indicator icon to the right hand side. This will signify that the date is past due
![Client List](https://user-images.githubusercontent.com/437360/103033688-e09ecd00-4517-11eb-9852-62e2b7c25539.png)

AlexK-ino commented:
Hello @ohaibbq, @jessex!
Thanks for the comment, I corrected all the above, I'm not sure what, I understood point # 10 correctly, but I corrected the shadows.

Regarding the point, I initially did not understand its purpose and therefore flew out of my head, I apologize.

The problem with the build was due to an accidental prettier rule change. I also apologize for this incident. I will try not to make similar mistakes in the future.

ohaibbq reviewed (approved):
@AlexK-ino This looks so good, nice work!

Let's get this merged. Could you review #24 and start work on #23  and #25?

jessex reviewed (approved):
Looks great, @AlexK-ino! Looks like there are some merge conflicts but don't block on me.

ohaibbq commented:
@AlexK-ino could you fix the merge conflicts and get this PR merged?

AlexK-ino commented:
@ohaibbq, I fixed the conflicts, but unfortunately, I cannot merge branches, there is no access
